src/e2ee/RustEngine.ts

```diff
@@ -90,9 +89,11 @@ export class RustEngine {
                 break;
             case "joined":
             default:
-            // Default and other cases handled by assignment before switch
+                memberships.splice(-1);
         }
 
+        const members = (await this.client.getRoomMembers(roomId, null, memberships)).map(u => new UserId(u.membershipFor));
```

Comment:
we should probably be using `getRoomMembersByMembership` so the set of invited users can fail without failing encryption.

Reply:
ccde51b